README.rst

```diff
@@ -110,147 +145,237 @@ version of the AWS Encryption SDK, we recommend using the default value.
-CMKs with which to encrypt and decrypt data. On encryption, it encrypts the plaintext with all
-configured CMKs. On decryption, it only attempts to decrypt ciphertexts that have been wrapped
-with a CMK that matches one of the configured CMK ARNs.
-
-To create a ``StrictAwsKmsMasterKeyProvider`` you must provide one or more CMKs. For providers that will only
-be used for encryption, you can use any valid `KMS key identifier`_. For providers that will be used for decryption, you
-must use the key ARN; key ids, alias names, and alias ARNs are not supported.
-
-Because the ``StrictAwsKmsMasterKeyProvider`` uses the `boto3 SDK`_ to interact with `AWS KMS`_,
+An ``AwsKmsKeyring`` is configured with an AWS KMS key ARN whose AWS KMS key
+will be used to generate, encrypt, and decrypt data keys.
+On encryption, it encrypts the plaintext with the data key.
+On decryption, it decrypts an encrypted version of the data key,
+then uses the decrypted data key to decrypt the ciphertext.
+
+To create an ``AwsKmsKeyring`` you must provide a AWS KMS key ARN.
+For keyrings that will only be used for encryption,
+you can use any valid `KMS key identifier`_.
+For providers that will be used for decryption,
+you must use the key ARN.
+Key ids, alias names, and alias ARNs are not supported for decryption.
+
+Because the ``AwsKmsKeyring`` uses the `boto3 SDK`_ to interact with `AWS KMS`_,
 it requires AWS Credentials.
 To provide these credentials, use the `standard means by which boto3 locates credentials`_ or provide a
-pre-existing instance of a ``botocore session`` to the ``StrictAwsKmsMasterKeyProvider``.
+pre-existing instance of a ``botocore session`` to the ``AwsKmsKeyring``.
 This latter option can be useful if you have an alternate way to store your AWS credentials or
 you want to reuse an existing instance of a botocore session in order to decrease startup costs.
 
-If you configure the the ``StrictAwsKmsMasterKeyProvider`` with multiple CMKs, the `final message`_
-will include a copy of the data key encrypted by each configured CMK.
-
 .. code:: python
 
+    import boto3
+    from aws_cryptographic_materialproviders.mpl import AwsCryptographicMaterialProviders
+    from aws_cryptographic_materialproviders.mpl.config import MaterialProvidersConfig
+    from aws_cryptographic_materialproviders.mpl.models import CreateAwsKmsKeyringInput
+    from aws_cryptographic_materialproviders.mpl.references import IKeyring
+
     import aws_encryption_sdk
+    from aws_encryption_sdk import CommitmentPolicy
 
-    kms_key_provider = aws_encryption_sdk.StrictAwsKmsMasterKeyProvider(key_ids=[
-        'arn:aws:kms:us-east-1:2222222222222:key/22222222-2222-2222-2222-222222222222',
-        'arn:aws:kms:us-east-1:3333333333333:key/33333333-3333-3333-3333-333333333333'
-    ])
+    # Instantiate the encryption SDK client.
+    client = aws_encryption_sdk.EncryptionSDKClient(
+        commitment_policy=CommitmentPolicy.REQUIRE_ENCRYPT_REQUIRE_DECRYPT
+    )
```

Comment:
```suggestion
```

Let's drop client creation.
If we try to go 1:1 with the master key provider examples, we see that the previous example didn't make a client.

Reply:
Now that I look at this again, can we actually just use an `AwsKmsMultiKeyring` for this first example instead?
I'm seeing that the first example uses the equivalent of the `AwsKmsMultiKeyring`, since you can configure it with multiple KMS keys.